pageserver/compaction/src/helpers.rs

```diff
@@ -171,8 +195,8 @@ where
                 match ready!(load_future.as_mut().poll(cx)) {
                     Ok(entries) => {
                         this.load_future.set(None);
-                        *this.heap.peek_mut().unwrap() =
-                            LazyLoadLayer::Loaded(VecDeque::from(entries));
+                        this.heap
+                            .push(LazyLoadLayer::Loaded(VecDeque::from(entries)));
```

Comment:
PeekMut will always recheck the ordering (it's equivalent to pop+push). I think the Rust docs is saying that if something stays in the heap, the ordering should not be changed, but it's not applying to peek mut.

Reply:
Do you have sources for that? I didn't see anything about that in the docs of either [peek_mut](https://doc.rust-lang.org/nightly/std/iter/struct.Peekable.html#method.peek_mut) nor of [PeekMut](https://doc.rust-lang.org/nightly/std/collections/binary_heap/struct.PeekMut.html), and looking at the source code it also doesn't seem like that.

Reply:
You have link to wrong `peek_mut` above. The correct peek_mut is this: https://doc.rust-lang.org/std/collections/struct.BinaryHeap.html#method.peek_mut. The example on that demonstrates modifying peeked the value.

The heap ordering is re-checked and fixed if needed by the 'sift_down' call at https://doc.rust-lang.org/src/alloc/collections/binary_heap/mod.rs.html#321.

Reply:
This is very subtle, it took me a while to understand how peek_mut() works. That's why I put the comment "(this decreaseKey operation is performed implicitly when `top` is dropped)" below

Reply:
I suppose then the concerns I had about the BinaryHeap are resolved and the code is correct I suppose. I will revert the commit.